src/engine/sync/enginesync.cpp

```diff
@@ -32,6 +32,57 @@ EngineSync::EngineSync(UserSettingsPointer pConfig)
 EngineSync::~EngineSync() {
 }
 
+Syncable* EngineSync::pickMaster(Syncable* enabling_syncable) {
+    std::vector<Syncable*> stopped_sync_decks;
+    std::vector<Syncable*> playing_sync_decks;
+
+    if (enabling_syncable != nullptr && enabling_syncable->getBaseBpm() != 0.0) {
+        if (enabling_syncable->isPlaying()) {
+            playing_sync_decks.push_back(enabling_syncable);
+        } else {
+            stopped_sync_decks.push_back(enabling_syncable);
+        }
+    }
+
+    for (const auto& pSyncable: m_syncables) {
+        if (pSyncable == enabling_syncable) {
+            continue;
+        }
+        // This is a string comparison, but this function is not hot.
```

Comment:
"hot" meaning "called a lot."  "Hot Path" or "hot function" refers to any function that may be called thousands or millions of times.  Since this is only called on master changes, and only once per syncable, it's not hot

Reply:
ok you're right it should be a bool.  I'm thinking m_bIsPrimaryDeck, and adding that to enginechannel.h.  We can set it false by default and the main channels will override it to true.  sound good?

Reply:
Perfect, if we have a comment about it as well.

Reply:
done